### ISSUE TITLE
Button: Add `type` attribute

### ISSUE THREAD

**Issue by coliff** [trimble-oss/modus-web-components#2167]
We should add the `type` attribute to the button element. The default should be `type="button"`, but user can also choose "`button`", "`submit`" or "`reset`".

REF: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#type

It's good practise to always specify the type because the default type is `submit` (not `button` as one might expect).

REF: https://webhint.io/docs/user-guide/hints/hint-button-type/?source=devtools
